Mathlib/MeasureTheory/Function/LpSeminorm/Basic.lean

```diff
@@ -1102,6 +1102,49 @@ theorem eLpNorm'_le_nnreal_smul_eLpNorm'_of_ae_le_mul {f : α → F} {g : α →
   simp_rw [ENNReal.coe_le_coe, ← NNReal.mul_rpow, NNReal.rpow_le_rpow_iff hp]
   exact h
 
+-- XXX: should the nnnorm version be deleted or primed instead?
+theorem eLpNorm'_le_nnreal_smul_eLpNorm'_of_ae_le_mul' {f : α → ε} {g : α → ε'} {c : ℝ≥0}
+    (h : ∀ᵐ x ∂μ, ‖f x‖ₑ ≤ c * ‖g x‖ₑ) {p : ℝ} (hp : 0 < p) :
+    eLpNorm' f p μ ≤ c • eLpNorm' g p μ := by
+  simp_rw [eLpNorm'_eq_lintegral_enorm]
+  rw [← ENNReal.rpow_le_rpow_iff hp, ENNReal.smul_def, smul_eq_mul,
+    ENNReal.mul_rpow_of_nonneg _ _ hp.le]
+  simp_rw [← ENNReal.rpow_mul, one_div, inv_mul_cancel₀ hp.ne.symm, ENNReal.rpow_one,
+    ← ENNReal.coe_rpow_of_nonneg _ hp.le, ← lintegral_const_mul' _ _ ENNReal.coe_ne_top]
+  apply lintegral_mono_ae
+  have aux (x) : (↑c) ^ p * ‖g x‖ₑ ^ p = (↑c * ‖g x‖ₑ) ^ p := by
+    have : ¬(p < 0) := by linarith
+    simp [ENNReal.mul_rpow_eq_ite, enorm_eq_zero, this]
+  simpa [ENNReal.coe_rpow_of_nonneg _ hp.le, aux, ENNReal.rpow_le_rpow_iff hp]
+
+variable {ε'' : Type*} [MeasurableSpace ε''] [TopologicalSpace ε'']
+  [ContinuousENorm ε''] [OpensMeasurableSpace ε''] in
+theorem eLpNorm'_le_mul_eLpNorm'_of_ae_le_mul {f : α → ε} {c : ℝ≥0∞} {g : α → ε''} {p : ℝ}
```

Comment:
This lemma should get a docstring, and a version for `eLpNorm` (no prime).
But do we actually need assumption `hg`? Why doesn't `lintegral_eq_zero_iff'` hold without assuming a.e.-measurability? I might be wrong here, but proof sketch:
* Suppose `f` is not a.e. `0` but has `lintegral` 0
* Then `{f != 0}` has positive measure
* Then `{f \ge 1 / n}` has positive measure for some natural `n`
* Contradiction by definition of `lintegral`.

There is probably something wrong here, but I don't see it.

Reply:
Also, isn't this relatively easy from [MeasureTheory.lintegral_mono_ae](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Integral/Lebesgue.html#MeasureTheory.lintegral_mono_ae)?
Note that the Lebesgue integral is *not* defined to be 0 for non-measurable functions. We have a much better default value: the supremum of the integral of all smaller measurable functions.

Reply:
I must say the docstrings could be improved to explain this better

Reply:
Would you also like me to add a (draft) counterexample PR formalising this - depending on the definition and properties of the Vitali set (from another in-progress PR)?

Reply:
I just created #23707 for the counterexamples PR - which I won't pursue further, given that too many sorries about the Vitali set (from another PR) are missing.

Reply:
I don't think it's necessary for formalize the counterexample.

Reply:
And added an eLpNorm version of this lemma (a few lines further below)